File: server/src/main/java/com/uiptv/util/UrlAttributeParser.java
package com.uiptv.util;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class UrlAttributeParser implements StalkerAttributeParser {
    private static final Pattern URL_PATTERN = Pattern.compile("(https?://\\S+)");

    @Override
    public String parse(String line) {
        Matcher m = URL_PATTERN.matcher(line);
        return m.find() ? m.group(1) : null;
    }

    @Override
    public StalkerAttributeType getAttributeType() {
        return StalkerAttributeType.URL;
    }
}

File: server/src/main/kotlin/com/uiptv/util/UrlAttributeParser.kt
package com.uiptv.util

import com.uiptv.api.StalkerAttributeParser
import java.util.regex.Pattern

class UrlAttributeParser : StalkerAttributeParser {
    override fun parse(line: String): String? {
        val matcher = URL_PATTERN.matcher(line)
        return if (matcher.find()) matcher.group(1) else null
    }

    override fun getAttributeType(): StalkerAttributeType = StalkerAttributeType.URL

    private companion object {
        val URL_PATTERN: Pattern = Pattern.compile("(https?://\\S+)")
    }
}
